dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -196,6 +212,59 @@ public void checkEntryPermissions(final Optional<User> user, final Entry<? exten
         }
     }
 
+    @POST
+    @Path("/{entryId}/exportToOrcid")
+    @Timed
+    @UnitOfWork
+    @Operation(description = "Export entry to ORCID", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiResponse(responseCode = HttpStatus.SC_NO_CONTENT + "", description = "No Content")
+    @ApiResponse(responseCode = HttpStatus.SC_INTERNAL_SERVER_ERROR + "", description = "Internal Server Error")
+    @ApiResponse(responseCode = HttpStatus.SC_NOT_FOUND + "", description = "Not Found")
+    @ApiResponse(responseCode = HttpStatus.SC_BAD_REQUEST + "", description = "Bad Request")
+    @ApiOperation(value = "hidden", hidden = true)
+    public void exportToORCID(@Parameter(hidden = true, name = "user") @Auth User user, @Parameter(description = "The id of the entry to export.", name = "entryId", in = ParameterIn.PATH, required = true)
+        @PathParam("entryId") Long entryId,
+        @Parameter(description = "Optional version ID of the entry version to export.", name = "versionId", in = ParameterIn.QUERY) @QueryParam("versionId") Long versionId) {
+        Entry<? extends Entry, ? extends Version> entry = toolDAO.getGenericEntryById(entryId);
+        checkEntry(entry);
+        checkEntryPermissions(Optional.of(user), entry);
+        List<Token> orcidByUserId = tokenDAO.findOrcidByUserId(user.getId());
+
+        Optional<Version> optionalVersion = Optional.empty();
+        if (versionId != null) {
+            Version version = versionDAO.findVersionInEntry(entry.getId(), versionId);
+            if (version == null) {
+                throw new CustomWebApplicationException("Version does not belong to entry", HttpStatus.SC_BAD_REQUEST);
+            }
+            if (version.getDoiURL() == null) {
+                throw new CustomWebApplicationException("Version does not have a DOI url associated with it", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Use constant strings here shared with the tests so it makes it obvious when changing one, the other gets changed